libraries/Microsoft.Bot.Builder/TurnContextServiceCollection.cs

```diff
@@ -14,30 +12,8 @@ namespace Microsoft.Bot.Builder
     /// <remarks>
     /// TODO: add more details on what kind of services can/should be stored here, by whom and what the lifetime semantics are, etc.
     /// </remarks>
-    public interface ITurnContextServiceCollection : IEnumerable<KeyValuePair<string, object>>
+    public sealed class TurnContextServiceCollection : Dictionary<string, object>, IDisposable
```

Comment:
Why is this sealed? Doesn't seem like it should be. Are we worried about someone subclassing and changing semanitcs?

Reply:
If sealed is the "Right" choice, please add a comment in the code as to why.